js/about/bookmarks.js

```diff
@@ -37,14 +37,17 @@ class BookmarkFolderItem extends ImmutableComponent {
 
   onDrop (e) {
     const bookmark = dndData.getDragData(e.dataTransfer, dragTypes.BOOKMARK)
+
     if (bookmark) {
+      // Don't allow a bookmark folder to be moved into itself
+      if (bookmark.get('folderId') === this.props.bookmarkFolder.get('folderId')) {
```

Comment:
we have another check somewhere that checks all decedents but I don't remember where off hand.   Is this a strong enough check? I.e. could someone do something like this?
```
A
--> B
     --- C
```
And then move A into C to create a circular ref?